pkg/reconciler/ingress/resources/virtual_service.go

```diff
@@ -118,7 +125,35 @@ func MakeVirtualServices(ia v1alpha1.IngressAccessor, gateways map[v1alpha1.Ingr
 	if requiredGatewayCount > 0 {
 		vss = append(vss, MakeIngressVirtualService(ia, gateways))
 	}
-	return vss
+
+	// Add probe routes
+	for _, vs := range vss {
+		hash, err := computeVirtualServiceHash(vs)
+		if err != nil {
+			return nil, fmt.Errorf("failed to compute VirtualService hash: %v", err)
```

Comment:
probably the name is useful as well, but i'd  actually go with `errors.Wrap(f)`.